debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbConnector.java

```diff
@@ -205,28 +93,35 @@ public ConfigDef config() {
     public Config validate(Map<String, String> connectorConfigs) {
         final Configuration config = Configuration.from(connectorConfigs);
 
-        // First, validate all of the individual fields, which is easy since don't make any of the fields invisible ...
-        Map<String, ConfigValue> results = validateAllFields(config);
-
-        // Get the config values for each of the connection-related fields ...
-        ConfigValue connectionStringValue = results.get(MongoDbConnectorConfig.CONNECTION_STRING.name());
-        ConfigValue userValue = results.get(MongoDbConnectorConfig.USER.name());
-        ConfigValue passwordValue = results.get(MongoDbConnectorConfig.PASSWORD.name());
-
-        // If there are no errors on any of these ...
-        if (userValue.errorMessages().isEmpty()
-                && passwordValue.errorMessages().isEmpty()
-                && connectionStringValue.errorMessages().isEmpty()) {
-            // Try to connect to the database ...
-            ConnectionContext connContext = new ConnectionContext(config);
-            try (MongoClient client = connContext.connect()) {
+        // Validate all fields and get connection string validation result
+        Map<String, ConfigValue> validation = validateAllFields(config);
+        ConfigValue csValidation = validation.get(MongoDbConnectorConfig.CONNECTION_STRING.name());
+
+        // Validate connection when connection string is otherwise valid
+        if (csValidation.errorMessages().isEmpty()) {
+            validateConnection(config, csValidation);
+        }
+        return new Config(new ArrayList<>(validation.values()));
+    }
+
+    public void validateConnection(Configuration config, ConfigValue connectionStringValidation) {
+        MongoDbConnectionContext connectionContext = new MongoDbConnectionContext(config);
+
+        try {
+            // Check base connection by accessing first database name
+            try (MongoClient client = connectionContext.getMongoClient()) {
                 client.listDatabaseNames().first(); // only when we try to fetch results a connection gets established
             }
-            catch (MongoException e) {
-                connectionStringValue.addErrorMessage("Unable to connect: " + e.getMessage());
+
+            // For RS clusters check that replica set name is present
+            // Java driver is smart enough to work without it but the specs says it should be set
+            if (!connectionContext.hasRequiredReplicaSetName()) {
```

Comment:
No.. but it can be removed (The one in `MongoUtil`). Missing ReplicaSet name for RS deployments now produces an error.. it's an easy fix and it leads users to use proper configuration.